### DISCUSSION TITLE
Any antialiasing when a sprite is scaled up at 2x?

### DISCUSSION THREAD

**Discussion by aleb** [bevyengine/bevy#4305] [Q&A] [answered]
In a 2D Bevy app we place SpriteBundles which are scaled 2x. Will each pixel result in 4 pixels of exactly the same color? Does antialiasing take place?

```
  let FACTOR: f32 = 2.0;

  commands
      .spawn_bundle(SpriteBundle {
          texture: asset,
          transform: Transform {
              scale: Vec3::ONE * FACTOR,
              ..Default::default()
          },
          ..Default::default()
      });
```

I believe this might be relevant―we're using the following startup system:
```
    commands.spawn_bundle(UiCameraBundle::default());
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
```

**Comment by superdump** [comment, marked as answer]
The `Image` `Asset` has a `SamplerDescriptor` that you can configure. By default it looks like it will do no filtering for minification/magnification, ie nearest neighbour.

**Reply by aleb** [reply]
`image.sampler_descriptor.mag_filter = FilterMode.NEAREST`

https://github.com/bevyengine/bevy/blob/release-0.6.1/crates/bevy_render/src/texture/image.rs#L29
https://github.com/gfx-rs/wgpu/blob/v0.12/wgpu/src/lib.rs#L1193
https://github.com/gfx-rs/wgpu/blob/v0.12/wgpu-types/src/lib.rs#L3078

**Reply by superdump** [reply]
Exactly.

One thing I did note after posting this was the the glTF loader will configure the samplers according to glTF defaults / whatever is specified in the glTF model.
